module_4/ex_4_6_5.py

```diff
@@ -0,0 +1,135 @@
+'''
+Consignas:
+    Escribir funciones que resuelvan los siguientes problemas:
+    a) Dado un año indicar si es bisiesto.
+    Nota: un año es bisiesto si es un número divisible por 4, pero no si es divisible por 100, excepto que también sea divisible por 400.
+    b) Dado un mes y un año, devolver la cantidad de días correspondientes.
+    c) Dada una fecha (día, mes, año), indicar si es válida o no.
+    d) Dada una fecha, indicar los días que faltan hasta fin de mes.
+    e) Dada una fecha, indicar los días que faltan hasta fin de año.
+    f) Dada una fecha, indicar la cantidad de días transcurridos en ese año hasta esa fecha.
+    g) Dadas dos fechas (día1, mes1, año1, día2, mes2, año2), indicar el tiempo transcurrido entre ambas, en años, meses y días.
+    Nota: en todos los casos, invocar las funciones escritas previamente cuando sea posible.
+'''
+MESES_DIAS = [0, 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]
+
+def es_bisiesto(y):
+    '''
+        Función que retorna True o False si un año es o no bisiesto.
+    '''
+    return y % 4 == 0 and y % 100 != 0 or y % 400 == 0
+        
+def contador_dias(m, y):
+    '''
+        Función que devuelve el total de días transcurridos hasta un mes determinado.
+    '''
+    total_dias = 0  
+    for x in range(1, m + 1):
+        total_dias+= MESES_DIAS[x]
+
+    if es_bisiesto(y) and m > 1:
+        total_dias+=1
+    return total_dias
+
+def evalua_fecha(d, m, y):
+    '''
+        Comprueba si una fecha pasada por parámetro es vádida.
+    '''
+    if  (d < 1 or d > 31) or (m < 1 or m > 12) or (y < 1):
+        return False
+    
+    if m == 2 and not es_bisiesto(y) and d > 28:
+        return False
+    
+    if m != 2 and d > MESES_DIAS[m]:
```

Comment:
siendo este el ultimo if y el retorno de la funcion es boolean podriamos retornar la condicion, no?

`return not (m != 2 and d > MESES_DIAS[m])`

lo negue porque si la condicion era True vos devolvias False